backend/src/layer/data/flickr_api/flickr_api_handler.rs

```diff
@@ -18,22 +18,23 @@ pub struct FlickrApiHandler {
 
 lazy_static! {
     static ref LONG_URL_REGEX: Regex =
-        Regex::new(r"(https://www.flickr.com/photos/)(\w+)/(\d+)([/]{0,1})")
+        Regex::new(r"(https://www.flickr.com/photos/)([\d\w@]+)/(\d+)([/]{0,1})")
             .expect("regex creation failed");
     static ref SHORT_URL_REGEX: Regex =
         Regex::new(r"(https://flic.kr/p/)([\d\w]+)").expect("regex creation failed");
 }
 
 impl FlickrApiHandler {
     #[must_use]
-    #[allow(clippy::missing_const_for_fn)] // cannot be made const because of compiler error`
+    #[allow(clippy::missing_const_for_fn)]
     pub fn new(info: FlickrInfo) -> Self {
         Self {
             request: ApiRequest::new(info.api_key),
         }
     }
 
-    // URL TYPE 1: https://www.flickr.com/photos/gerdavs/52310534489/ <- remove last '/'
+    // URL TYPE 1.1: https://www.flickr.com/photos/gerdavs/52310534489/ <- remove last '/'
+    // URL TYPE 1.2: https://www.flickr.com/photos/198319418@N06/53077317043 <- remove last '/'
     // URL TYPE 2: https://flic.kr/p/2oRguN3
     // Both cases: Split with '/' and get last member (= photo_id).
     fn determine_photo_id(mut url: &str) -> Result<&str> {
```

Comment:
why did you go away from using regex capute roups to get the id?
This would be more reliable and easier to adapt

Reply:
i want to reduce regex checks in the function (like `if matches1 || matches2 ...`)